### PR TITLE
Target audience tab

### PR DESCRIPTION
**Pull request by miri**
#70

### CODE REVIEW

**Comment by miri** [comment]
Clean PR from PR 130.

File: haskala/modules/custom/haskala_book/haskala_book.module [hunk 1]
```diff
@@ -62,24 +62,57 @@ function haskala_book_get_production($book_nid, $person_nid = NULL) {
...
-    if ($wrapper->$field_name->value()){
-      $weight = $node_view[$field_name]['#weight'];
-      $tab_fields_sorted[$weight] = $field_name;
+
+  foreach($tab->children as $child_name) {
+
+    // If exist sub group. Ex.:Target audience tab contains sub groups.
+    if (strpos($child_name, 'group') !== FALSE) {
+      $sub_tab = field_group_load_field_group($child_name, 'node', 'book', 'default');
+      // Recorsive call
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/haskala_book.module]: Recursive call.

File: haskala/modules/custom/haskala_book/haskala_book.module [hunk 2]
```diff
@@ -62,24 +62,57 @@ function haskala_book_get_production($book_nid, $person_nid = NULL) {
...
   $tab_fields_sorted = array();
-  foreach($tab->children as $field_name) {
-    if ($wrapper->$field_name->value()){
-      $weight = $node_view[$field_name]['#weight'];
-      $tab_fields_sorted[$weight] = $field_name;
+
+  foreach($tab->children as $child_name) {
+
+    // If exist sub group. Ex.:Target audience tab contains sub groups.
+    if (strpos($child_name, 'group') !== FALSE) {
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/haskala_book.module]: Check with substr() for names starting with 'group'

File: haskala/modules/custom/haskala_book/haskala_book.module [hunk 3]
```diff
@@ -62,24 +62,57 @@ function haskala_book_get_production($book_nid, $person_nid = NULL) {
...
+  }
 
   // Sort fields by weight.
   $tab_fields_sorted = array();
-  foreach($tab->children as $field_name) {
-    if ($wrapper->$field_name->value()){
-      $weight = $node_view[$field_name]['#weight'];
-      $tab_fields_sorted[$weight] = $field_name;
+
+  foreach($tab->children as $child_name) {
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/haskala_book.module]: foreach (

File: haskala/modules/custom/haskala_book/haskala_book.module [hunk 4]
```diff
@@ -62,24 +62,57 @@ function haskala_book_get_production($book_nid, $person_nid = NULL) {
...
+      // Recorsive call
+      $sub_tabs_fields[$child_name] = haskala_book_get_tab_fields($wrapper, $sub_tab, $node_view, $book_nid);
+      if ($sub_tabs_fields[$child_name]) {
+        // Add sub title at the beginning of the array.
+        $sub_tabs_fields[$child_name] = array('sub_title' => $sub_tab->label) + $sub_tabs_fields[$child_name];
+
+        $sub_tabs_fields[$child_name]['weight'] = $sub_tab->weight;
+      }
+    }
+    else {
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/haskala_book.module]: elseif ($wrapper->$child_name->value()) {

File: haskala/modules/custom/haskala_book/plugins/content_types/book_fields/haskala-book-fields.tpl.php [hunk 5]
```diff
@@ -51,8 +51,15 @@
               </div>
             <?php else: ?>
               <div class="fields-multi">
-                <?php foreach($single_tab_fields as $tab_field): ?>
-                  <?php print render($tab_field); ?>
+                <?php foreach($single_tab_fields as $key_field => $tab_field): ?>
+                  <?php // Sub group in tab if exist.
+                  if ($key_field === 'sub_title'): ?>
+                    <div class = "tab-sub-title">
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/plugins/content_types/book_fields/haskala-book-fields.tpl.php]: class="

Comment (bitamar) [inline haskala/modules/custom/haskala_book/plugins/content_types/book_fields/haskala-book-fields.tpl.php]: class="

**Comment by bitamar** [comment]
Merged

**Pull request closed without merging**